packages/admin/cms-admin/src/dam/Table/fileUpload/useFileUpload.tsx

```diff
@@ -192,15 +197,61 @@ export const useFileUpload = (options: UploadFileOptions): FileUploadApi => {
         [client],
     );
 
+    const createInitialFolderIdMap = React.useCallback(
```

Comment:
The folderIdMap holds all ids of existing folders that the user is trying to upload files to.

Previously, the folderIdMap was created entirely on the fly during the upload. Now, that's not possible anymore because I have to check if a filename is occupied or not before the upload. Therefore, I now create the folderIdMap in the beginning. It's subsequently filled with all ids of existing folders. 

Folders that are created during the upload are still added to the folderIdMap on the fly. (That's possible because they are irrelevant for the isFilenameOccupied check)